File: Day_11/CodeWars.py
# 1) Is Palindrome:

def is_palindrome(str = ""):
    if len(str) == 0:
        return True
    for letter in range(0, len(str)):
        if str.upper() == str[::-1].upper(): #testing index 0 to 7-(0+1)= 6
            return True

        return False

is_palindrome("raceCar")


# 2) Counting Sheep:

array1 = [True,  True,  True,  False,
True,  True,  True,  True ,
True,  False, True,  False,
True,  False, False, True ,
True,  True,  True,  True ,
False, False, True,  True ]


def count_sheeps(li=[]):
    count = 0
    for x in li:
        if x == True:
            count += 1
    return count

# print(count_sheeps(array1))


# 3) Convert a Boolean to a String:

def boolean_to_string(boolean):
    b = str(boolean)
    print(type(b))

# boolean_to_string(True)


# 4) Count number of cubes with paint on:

def count_squares(y):
    x = (((y+1)**2)*2) + ((y + 1)*(y - 1)* 2) + (((y-1)**2)*2)
    print(x)

count_squares(5)

File: Day_11/CodeWars.js
// 1) Is Palindrome:

function isPalindrome(str){
    // Use .split to convert the string into an array 
    // the "" split every letter into an array.
    var li1 = str.split("");
    console.log(li1)
    // Use .reverse() to reverse the array.
    var li2 = li1.reverse();
    console.log(li2)
    // Use .join() to make the reversed array items into a string. 
    var str2 = li2.join("");
    console.log(str2)
    // Compare both strings.
    if (str.toUpperCase() == str2.toUpperCase()){
        return true;
    }
    else {
        return false;
    }
}
// console.log(isPalindrome("aba"))

// Elvis' Solution:
// for(var i = x.length - 1; i >= 0; i--){
//     backwardStr += x[i];
// }
//  //Example 1: "sivlE" equal "Elvis" -> false
//  //Example 2: "racecar" equal "racecar" -> true

// if(backwardStr.toUpperCase() == x.toUpperCase()){
//     return true;
// }
// else{
//     return false;
// }


// 2) Counting Sheep:



function countSheeps(arrayOfSheep) {
    var count = 0
    for (var i= 0; (arrayOfSheep.length); i++){
        if (arrayOfSheep[i] == true ){
            count++;
        }
    }
}

// console.log(countSheeps())


// 3) Convert a Boolean to a String:

function booleanToString(Boolean){
    var b = Boolean.toString();
    console.log(b)
    console.log(typeof(b))
    return b
}

booleanToString(true)

// 4) Count number of cubes with paint on:

var countSquares = function(cuts){  
    var countSquares = (((y+1)**2)*2) + ((y + 1)*(y - 1)* 2) + (((y-1)**2)*2)

}

function countSquares(cuts){
    var countSquares = (((cuts+1)**2)*2) + ((cuts + 1)*(cuts - 1)* 2) + (((cuts-1)**2)*2);
    return countSquares;

}